packages/ssz/src/core/createValue.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import assert from "assert";
+import {AnySSZType, FullSSZType, parseType, Type} from "@chainsafe/ssz-type-schema";
+
+import {assertValidValue} from "./assertValidValue";
+import {defaultValue} from "./defaultValue";
+
+/**
+ * Given an ssz type, create an empty value, optionally filled in with a default value
+ *
+ * The default value can be only partially specified.
+ */
+export function createValue<T>(type: AnySSZType<T>, value: any = null): T {
+  return _createValue(parseType(type), value) as T;
+}
+
+function _createValue(type: FullSSZType, value: any = null): any {
+  switch(type.type) {
+    case Type.uint:
+    case Type.bool:
+    case Type.bitList:
+    case Type.bitVector:
+    case Type.byteList:
+    case Type.byteVector:
+      if (value !== null && value !== undefined) {
+        assertValidValue(value, type);
+        return value;
+      } else {
+        return defaultValue(type);
+      }
+    case Type.list:
+      if (value) {
+        assert(Array.isArray(value));
+      } else {
+        value = [];
```

Comment:
What cases would this be hit? Not seeing it.

Reply:
if `value === null`

Reply:
Oh yeah, youre right. I misread :\

Reply:
Should you use the default value then?

Reply:
The function that you created

Reply:
Oh yeah that would be better